src/gluonts/evaluation/new_metrics.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass
+from functools import partial
+from itertools import chain
+from typing import List
+
+import numpy as np
+
+
+class Stat:
+    name: str
+    dependencies = ()
+
+    def apply(self, x):
+        raise NotImplementedError
+
+
+class Metric:
+    name: str
+    dependencies = ()
+
+    def apply(self, x):
+        x[self.name] = self.loss(x)
+
+    def get_aggr(self, metrics, aggrs, axis):
+        return self.aggregate(metrics[self.name], axis=axis)
+
+    def loss(self, x):
+        raise NotImplementedError
+
+    def aggregate(self, metrics, axis=None):
+        raise NotImplementedError
+
+
+class DerivedMetric:
```

Comment:
It seems that the term `DerivedMetric` corresponds to metrics that are available only for the aggregated cases, and not for item-wise cases. Is this interpretation correct?